src/python/pants/backend/python/macros/poetry_requirements.py

```diff
@@ -493,6 +496,8 @@ def generate_tgt(parsed_req: PipRequirement) -> PythonRequirementTarget:
                 **tgt_overrides,
             },
             request.template_address.create_generated(parsed_req.project_name),
+            name_explicitly_set=request.generator_name_explicitly_set,
+            union_membership=union_membership,
```

Comment:
I don't think that it is necessary to propagate this value to generated targets, because the warning is always at the generator level. When resolving an address, we first find the relevant generator. And so we can always warn by looking at the generator rather than the generated.